src/encode.rs

```diff
@@ -346,15 +345,28 @@ where
 }
 
 #[cfg(feature = "check")]
-fn encode_check_into(input: &[u8], output: &mut [u8], alpha: &[u8; 58]) -> Result<usize> {
+fn encode_check_into<'a, I>(
+    input: &[u8],
+    output: &mut [u8],
+    alpha: &[u8; 58],
+    version: I,
```

Comment:
If you keep this as `version: Option<u8>` then later call `encode_into(version.as_ref().into_iter().chain(...), ...)` I think this would work.

Reply:
Not sure why I did not try such way. Never mind, now it works. Thanks!